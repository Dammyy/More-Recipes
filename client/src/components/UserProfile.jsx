import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Navbar from './Navbar';
import {
  BtnHome,
  BtnAdd,
  BtnManageRecipes,
  BtnFavorites,
  BtnCatalog
} from './Buttons';

/**
 *
 *
 * @class Reviews
 * @extends {PureComponent}
 */
class UserProfile extends PureComponent {
  /**
   *
   * @memberOf UserProfile
   * @returns {void}
   *
   */
  render() {
    if (!this.props.user) {
      return <h1>Profile Loading...</h1>;
    }
    const { user } = this.props;
    return (
      <div>
        <Navbar />
        <div className="text-left-buttons btn-buttons">
          <BtnHome />
          <BtnCatalog />
          <BtnAdd />
          <BtnManageRecipes />
          <BtnFavorites />
        </div>
        <div className="profile-display">
          <div className="profile-heading">
            <h2>
              <i className="fa fa-user-o" aria-hidden="true" /> My Details
            </h2>
          </div>
          <p><b>First name:</b> {user.firstName}</p>
          <p><b>Surname:</b> {user.lastName}</p>
          <p><b>Email:</b> {user.email}</p>
          <div className="profile-heading">
            <h2>
              <i className="fa fa-line-chart" aria-hidden="true" /> My Stats
            </h2>
          </div>
          <p><b>Recipes published - </b> {user.recipeCount}</p>
          <p><b>Favorite recipes - </b> {user.favsCount}</p>
        </div>
      </div>
    );
  }
}

UserProfile.propTypes = {
  user: PropTypes.oneOfType([PropTypes.array, PropTypes.object]).isRequired,
};

export default UserProfile;
