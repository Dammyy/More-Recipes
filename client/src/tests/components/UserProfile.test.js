import React from 'react';
import { shallow } from 'enzyme';
import UserProfile from '../../components/UserProfile';

describe('UserProfile component', () => {
  describe('Snapshot', () => {
    test('component should fully render', () => {
      const props = {
        user: {}
      };
      const component = shallow(<UserProfile {...props} />);
      expect(component).toMatchSnapshot();
    });
  });
});

describe('UserProfile component', () => {
  describe('Snapshot', () => {
    test('component should fully render', () => {
      const component = shallow(<UserProfile />);
      expect(component).toMatchSnapshot();
    });
  });
});
