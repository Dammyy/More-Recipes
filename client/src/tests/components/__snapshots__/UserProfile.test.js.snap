// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile component Snapshot component should fully render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserProfile
    user={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ReduxForm />,
        <div
          className="text-left-buttons btn-buttons"
        >
          <BtnHome />
          <BtnCatalog />
          <Connect(authRecipeButtons(Component)) />
          <Connect(authRecipeButtons(Component)) />
          <Connect(authRecipeButtons(Component)) />
        </div>,
        <div
          className="profile-display"
        >
          <div
            className="profile-heading"
          >
            <h2>
              <i
                aria-hidden="true"
                className="fa fa-user-o"
              />
               My Details
            </h2>
          </div>
          <p>
            <b>
              First name:
            </b>
             
          </p>
          <p>
            <b>
              Surname:
            </b>
             
          </p>
          <p>
            <b>
              Email:
            </b>
             
          </p>
          <div
            className="profile-heading"
          >
            <h2>
              <i
                aria-hidden="true"
                className="fa fa-line-chart"
              />
               My Stats
            </h2>
          </div>
          <p>
            <b>
              Recipes published - 
            </b>
             
          </p>
          <p>
            <b>
              Favorite recipes - 
            </b>
             
          </p>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <BtnHome />,
            <BtnCatalog />,
            <Connect(authRecipeButtons(Component)) />,
            <Connect(authRecipeButtons(Component)) />,
            <Connect(authRecipeButtons(Component)) />,
          ],
          "className": "text-left-buttons btn-buttons",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="profile-heading"
            >
              <h2>
                <i
                  aria-hidden="true"
                  className="fa fa-user-o"
                />
                 My Details
              </h2>
            </div>,
            <p>
              <b>
                First name:
              </b>
               
            </p>,
            <p>
              <b>
                Surname:
              </b>
               
            </p>,
            <p>
              <b>
                Email:
              </b>
               
            </p>,
            <div
              className="profile-heading"
            >
              <h2>
                <i
                  aria-hidden="true"
                  className="fa fa-line-chart"
                />
                 My Stats
              </h2>
            </div>,
            <p>
              <b>
                Recipes published - 
              </b>
               
            </p>,
            <p>
              <b>
                Favorite recipes - 
              </b>
               
            </p>,
          ],
          "className": "profile-display",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h2>
                <i
                  aria-hidden="true"
                  className="fa fa-user-o"
                />
                 My Details
              </h2>,
              "className": "profile-heading",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    aria-hidden="true"
                    className="fa fa-user-o"
                  />,
                  " My Details",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-hidden": "true",
                    "className": "fa fa-user-o",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                " My Details",
              ],
              "type": "h2",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <b>
                  First name:
                </b>,
                " ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "First name:",
                },
                "ref": null,
                "rendered": "First name:",
                "type": "b",
              },
              " ",
              undefined,
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <b>
                  Surname:
                </b>,
                " ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Surname:",
                },
                "ref": null,
                "rendered": "Surname:",
                "type": "b",
              },
              " ",
              undefined,
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <b>
                  Email:
                </b>,
                " ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Email:",
                },
                "ref": null,
                "rendered": "Email:",
                "type": "b",
              },
              " ",
              undefined,
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h2>
                <i
                  aria-hidden="true"
                  className="fa fa-line-chart"
                />
                 My Stats
              </h2>,
              "className": "profile-heading",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    aria-hidden="true"
                    className="fa fa-line-chart"
                  />,
                  " My Stats",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-hidden": "true",
                    "className": "fa fa-line-chart",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                " My Stats",
              ],
              "type": "h2",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <b>
                  Recipes published - 
                </b>,
                " ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Recipes published - ",
                },
                "ref": null,
                "rendered": "Recipes published - ",
                "type": "b",
              },
              " ",
              undefined,
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <b>
                  Favorite recipes - 
                </b>,
                " ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Favorite recipes - ",
                },
                "ref": null,
                "rendered": "Favorite recipes - ",
                "type": "b",
              },
              " ",
              undefined,
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ReduxForm />,
          <div
            className="text-left-buttons btn-buttons"
          >
            <BtnHome />
            <BtnCatalog />
            <Connect(authRecipeButtons(Component)) />
            <Connect(authRecipeButtons(Component)) />
            <Connect(authRecipeButtons(Component)) />
          </div>,
          <div
            className="profile-display"
          >
            <div
              className="profile-heading"
            >
              <h2>
                <i
                  aria-hidden="true"
                  className="fa fa-user-o"
                />
                 My Details
              </h2>
            </div>
            <p>
              <b>
                First name:
              </b>
               
            </p>
            <p>
              <b>
                Surname:
              </b>
               
            </p>
            <p>
              <b>
                Email:
              </b>
               
            </p>
            <div
              className="profile-heading"
            >
              <h2>
                <i
                  aria-hidden="true"
                  className="fa fa-line-chart"
                />
                 My Stats
              </h2>
            </div>
            <p>
              <b>
                Recipes published - 
              </b>
               
            </p>
            <p>
              <b>
                Favorite recipes - 
              </b>
               
            </p>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <BtnHome />,
              <BtnCatalog />,
              <Connect(authRecipeButtons(Component)) />,
              <Connect(authRecipeButtons(Component)) />,
              <Connect(authRecipeButtons(Component)) />,
            ],
            "className": "text-left-buttons btn-buttons",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="profile-heading"
              >
                <h2>
                  <i
                    aria-hidden="true"
                    className="fa fa-user-o"
                  />
                   My Details
                </h2>
              </div>,
              <p>
                <b>
                  First name:
                </b>
                 
              </p>,
              <p>
                <b>
                  Surname:
                </b>
                 
              </p>,
              <p>
                <b>
                  Email:
                </b>
                 
              </p>,
              <div
                className="profile-heading"
              >
                <h2>
                  <i
                    aria-hidden="true"
                    className="fa fa-line-chart"
                  />
                   My Stats
                </h2>
              </div>,
              <p>
                <b>
                  Recipes published - 
                </b>
                 
              </p>,
              <p>
                <b>
                  Favorite recipes - 
                </b>
                 
              </p>,
            ],
            "className": "profile-display",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h2>
                  <i
                    aria-hidden="true"
                    className="fa fa-user-o"
                  />
                   My Details
                </h2>,
                "className": "profile-heading",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <i
                      aria-hidden="true"
                      className="fa fa-user-o"
                    />,
                    " My Details",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "aria-hidden": "true",
                      "className": "fa fa-user-o",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  " My Details",
                ],
                "type": "h2",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <b>
                    First name:
                  </b>,
                  " ",
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "First name:",
                  },
                  "ref": null,
                  "rendered": "First name:",
                  "type": "b",
                },
                " ",
                undefined,
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <b>
                    Surname:
                  </b>,
                  " ",
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Surname:",
                  },
                  "ref": null,
                  "rendered": "Surname:",
                  "type": "b",
                },
                " ",
                undefined,
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <b>
                    Email:
                  </b>,
                  " ",
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Email:",
                  },
                  "ref": null,
                  "rendered": "Email:",
                  "type": "b",
                },
                " ",
                undefined,
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h2>
                  <i
                    aria-hidden="true"
                    className="fa fa-line-chart"
                  />
                   My Stats
                </h2>,
                "className": "profile-heading",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <i
                      aria-hidden="true"
                      className="fa fa-line-chart"
                    />,
                    " My Stats",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "aria-hidden": "true",
                      "className": "fa fa-line-chart",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  " My Stats",
                ],
                "type": "h2",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <b>
                    Recipes published - 
                  </b>,
                  " ",
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Recipes published - ",
                  },
                  "ref": null,
                  "rendered": "Recipes published - ",
                  "type": "b",
                },
                " ",
                undefined,
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <b>
                    Favorite recipes - 
                  </b>,
                  " ",
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Favorite recipes - ",
                  },
                  "ref": null,
                  "rendered": "Favorite recipes - ",
                  "type": "b",
                },
                " ",
                undefined,
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`UserProfile component Snapshot component should fully render 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserProfile />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Profile Loading...",
    },
    "ref": null,
    "rendered": "Profile Loading...",
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Profile Loading...",
      },
      "ref": null,
      "rendered": "Profile Loading...",
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
